blas/tpls/KokkosBlas1_nrm1_tpl_spec_decl.hpp

```diff
@@ -207,40 +207,64 @@ KOKKOSBLAS1_CNRM1_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::HostSpace,
+        R.data()));
+  }
+  KOKKOS_CUBLAS_SAFE_CALL_IMPL(cublasSetStream(s.handle, NULL));
+}
+
+#define KOKKOSBLAS1_NRM1_TPL_SPEC_DECL_CUBLAS(SCALAR, LAYOUT, EXECSPACE,       \
+                                              MEMSPACE, ETI_SPEC_AVAIL)        \
   template <>                                                                  \
   struct Nrm1<                                                                 \
       EXECSPACE,                                                               \
-      Kokkos::View<double, LAYOUT, Kokkos::HostSpace,                          \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
-      Kokkos::View<const double*, LAYOUT, Kokkos::Device<EXECSPACE, MEMSPACE>, \
-                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<typename Kokkos::ArithTraits<SCALAR>::mag_type, LAYOUT,     \
+                   Kokkos::HostSpace,                                          \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
+      Kokkos::View<const SCALAR*, LAYOUT, Kokkos::Device<EXECSPACE, MEMSPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,                   \
       1, true, ETI_SPEC_AVAIL> {                                               \
     using execution_space = EXECSPACE;                                         \
-    typedef Kokkos::View<double, LAYOUT, Kokkos::HostSpace,                    \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        RV;                                                                    \
-    typedef Kokkos::View<const double*, LAYOUT,                                \
-                         Kokkos::Device<EXECSPACE, MEMSPACE>,                  \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        XV;                                                                    \
-    typedef typename XV::size_type size_type;                                  \
+    using RV = Kokkos::View<typename Kokkos::ArithTraits<SCALAR>::mag_type,    \
+                            LAYOUT, Kokkos::HostSpace,                         \
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>;          \
+    using XV = Kokkos::View<const SCALAR*, LAYOUT,                             \
+                            Kokkos::Device<EXECSPACE, MEMSPACE>,               \
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>;          \
+    using size_type = typename XV::size_type;                                  \
                                                                                \
     static void nrm1(const execution_space& space, RV& R, const XV& X) {       \
-      Kokkos::Profiling::pushRegion("KokkosBlas::nrm1[TPL_CUBLAS,double]");    \
+      Kokkos::Profiling::pushRegion("KokkosBlas::nrm1[TPL_CUBLAS,SCALAR]");    \
```

Comment:
`SCALAR` will not be substituted by the preprocessor here. (`##`? ScalarTraits?)

Reply:
I think a simple # would do the trick here... something like:
```
Kokkos::Profiling::pushRegion("KokkosBlas::nrm1[TPL_CUBLAS,"#SCALAR"]");
```
checking on weaver now and will move on to doing the same refactor for rocBLAS on caraway...